core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -119,6 +119,9 @@ private TableProperties() {
   public static final String DELETE_AVRO_COMPRESSION = "write.delete.avro.compression-codec";
   public static final String AVRO_COMPRESSION_DEFAULT = "gzip";
 
+  public static final String AVRO_ZSTD_COMPRESSION_LEVEL = "write.avro.zstd.compression-level";
```

Comment:
I think compression level can also be useful for gzip, we can just `write.avro.compression-level`

Reply:
actually, parquet also uses `write.parquet.compression-level`, so we can make it consistent here.

Reply:
and we also need a `write.delete.avro.compression-level`

Reply:
Yeah this is much cleaner. It looks like parquet just defines a common compression level property used for all compression codecs.

Reply:
Yes, and then we can pass just the compression level into `toCodec` and not all the properties.